src/lib.rs

```diff
@@ -267,6 +267,39 @@ fn run_tests_threaded<D: 'static + Send + Sync>(
     recv
 }
 
+/// Run the given tests one by one in the current thread.
+fn run_tests_serialized<D: 'static + Send + Sync>(
+    args: &Arguments,
+    tests: Vec<Test<D>>,
+    run_test: impl Fn(&Test<D>) -> Outcome + 'static + Send + Sync,
+) -> impl IntoIterator<Item = RunnerEvent<D>> {
+    let args_ignored = args.ignored;
+    let args_test = args.test;
+    let args_bench = args.bench;
+    let run_test = std::sync::Arc::new(run_test);
+    tests.into_iter().flat_map(move |test| {
+        let run_test = run_test.clone();
```

Comment:
Unpacking `args` and wrapping `run_test` in `Arc` is not necessary. 

```suggestion
fn run_tests_serialized<'a, D: 'static + Send + Sync>(
    args: &'a Arguments,
    tests: Vec<Test<D>>,
    run_test: &'a (impl Fn(&Test<D>) -> Outcome + 'static + Send + Sync),
) -> impl 'a + IntoIterator<Item = RunnerEvent<D>> {
    tests.into_iter().flat_map(move |test| {
```

Additionally, change the `args_ignored` below back to `args.ignored` (same for the other two). And finally, when calling this function pass `&run_tests` instead of `run_tests`.